### PR TITLE
added basic structure for vault role resource

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2362.

### CODE REVIEW

File: service/resource/vaultrolev1/update.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package vaultrolev1
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
```
Comment (JosephSalisbury) [inline service/resource/vaultrolev1/update.go]: Nitpick, these methods are in a different order than with delete :D

Comment (xh3b4sd) [reply service/resource/vaultrolev1/update.go]: Because it is alphabetical.

Comment (JosephSalisbury) [reply service/resource/vaultrolev1/update.go]: It is not alphabetical here - https://github.com/giantswarm/cert-operator/pull/144/files#diff-625498a136974b78c4f5da9492a042f9R10

Comment (xh3b4sd) [reply service/resource/vaultrolev1/update.go]: I see. Cool. I will fix this in the next PRs.

Comment (JosephSalisbury) [reply service/resource/vaultrolev1/update.go]: No stress man, I just know you like this stuff ordered <3

Comment (xh3b4sd) [reply service/resource/vaultrolev1/update.go]: I do and I will order the shit out of this man.

**Pull request merged**